crates/exex/src/serde.rs

```diff
@@ -1254,4 +1345,209 @@ mod tests {
             Err(KakarotSerdeError::InvalidFieldValue { field: "is_some".to_string() })
         );
     }
+
+    #[test]
+    fn test_serialize_dict_empty() {
+        // Setup
+        let mut kakarot_serde = setup_kakarot_serde(&TestProgram::KeccakAddUint256);
+
+        // Add a memory segment for the dictionary
+        let dict_ptr = kakarot_serde.runner.vm.add_memory_segment();
+
+        // Call serialize_dict with nothing in the dictionary
+        let result_dict_size = kakarot_serde
+            .serialize_dict(dict_ptr, None, Some(0))
+            .expect("failed to serialize dict");
+
+        // The result should be empty
+        assert!(result_dict_size.is_empty());
+
+        // Call serialize_dict with no members and no dict size
+        let result_no_dict_size =
+            kakarot_serde.serialize_dict(dict_ptr, None, None).expect("failed to serialize dict");
+
+        // The result should also be empty
+        assert!(result_no_dict_size.is_empty());
+    }
+
+    #[test]
+    fn test_serialize_dict_with_values() {
+        // Setup
+        let mut kakarot_serde = setup_kakarot_serde(&TestProgram::KeccakAddUint256);
+
+        // Add a memory segment for the dictionary
+        let dict_ptr = kakarot_serde.runner.vm.add_memory_segment();
+
+        // Insert key-value pairs into memory
+        let key1 = MaybeRelocatable::Int(Felt252::from(1));
+        let key2 = MaybeRelocatable::Int(Felt252::from(2));
+        let key3 = MaybeRelocatable::Int(Felt252::from(3));
+        let value1 = MaybeRelocatable::Int(Felt252::from(10));
+        let value2 = MaybeRelocatable::Int(Felt252::from(20));
+        let value3 =
+            MaybeRelocatable::RelocatableValue(Relocatable { segment_index: 10, offset: 15 });
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 0usize).unwrap(), key1.clone())
+            .expect("failed to insert key1");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 2usize).unwrap(), value1.clone())
+            .expect("failed to insert value1");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 3usize).unwrap(), key2.clone())
+            .expect("failed to insert key2");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 5usize).unwrap(), value2.clone())
+            .expect("failed to insert value2");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 6usize).unwrap(), key3.clone())
+            .expect("failed to insert key2");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 8usize).unwrap(), value3.clone())
+            .expect("failed to insert value2");
```

Comment:
insert directly a `Vec`, no need to do this 6 times

Reply:
ah yes forgot this trick again, fixed